core/src/main/java/org/apache/accumulo/core/client/sample/Sampler.java

```diff
@@ -61,4 +61,11 @@
    *         Return false if it should not be included.
    */
   boolean accept(Key k);
+
+  /**
+   * @param option
+   *          Option to validate.
+   * @return True if option is a valid sampler option. False otherwise.
+   */
+  boolean isValidOption(String option);
```

Comment:
Good idea, I will look into adding that improvement.

Reply:
Is there anything we should do for the default case? I currently have it verified to not be null. Not sure what else to add to the interface itself.

Reply:
The default implementation should probably be a no-op. It is assumed that the map of all the options will be non-null, because Accumulo would be responsible for preparing that map itself, but it's fine if you want to check for that, to be sure.

Reply:
I went ahead and kept the check. If the user forgets to even mention the option before compacting, it will cause the shell to hang.

Reply:
Nevermind. You are correct. I just realized the null gets thrown before reaching my validation check.

Reply:
Made these changes in my recent commit.